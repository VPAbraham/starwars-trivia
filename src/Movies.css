.section--movie-list {
  align-items: right;
  display: flex;
  overflow: auto;
  white-space: nowrap;
}

.ul--movie-list {
  background-color: rgb(90, 104, 208, .5);
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  list-style: none;
}

.li--movie {
  margin: 15px;
  text-align: center;
  width: 30vw;

}