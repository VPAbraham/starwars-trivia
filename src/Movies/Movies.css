.section--character-list,
.section--movie-list {
  align-items: right;
  background-color: rgb(90, 104, 208, .5);
  color: #ffffff;
  display: flex;
  font-size: 2.25rem;
  overflow: auto;
  white-space: nowrap;
}

.ul--character-list,
.ul--movie-list {
  display: flex;
  flex-direction: row;
  list-style: none;
}

.li--character,
.li--movie {
  margin: 15px;
  text-align: center;
  width: 30vw;
}