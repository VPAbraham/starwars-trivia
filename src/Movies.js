import React from 'react';
import './Movies.css';



const Movies = (props) => {
  const movieCards = props.map(movie => {
    <li>
      <MovieCard
        title={movie.title}
        episodeID={movie.episodeId}
        releaseDate={movie.releaseDate}
        characters={movie.characters}
        openingCrawl={movie.openingCrawl}
      />
    </li>  
  })
  return (
    <section>
      <ul>
        {movieCards}
      </ul>  
    </section>
  )
}



export default Movies;