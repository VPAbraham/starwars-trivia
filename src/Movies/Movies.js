import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './Movies.css';



const Movies = (props) => {
  const movieCards = props.movies.map(movie => {
    return (<li className='li li--movie'>
      <MovieCard
        key={movie.episodeId}
        title={movie.title}
        episodeId={movie.episodeId}
        releaseDate={movie.releaseDate}
        characters={movie.characters}
        openingCrawl={movie.openingCrawl}
      />
    </li>)  
  })
  return (
    <section className='section section--movie-list'>
      <ul className='ul ul--movie-list'>
        {movieCards}
      </ul>  
    </section>
  )
}



export default Movies;