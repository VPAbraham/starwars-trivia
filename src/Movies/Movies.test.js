import React from 'react';
import { shallow } from 'enzyme';
import Movies from './Movies';

describe('Movie', () => {
  const getCurrentMovieChars = jest.fn();
  it('should match snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
    <Movies
    movies={[]}
    getCurrentMovieChars={getCurrentMovieChars}
    />)
    expect(wrapper).toMatchSnapshot()
  });

  it('should be able to set the currently selected movie\'s characters to state', () => {
    const wrapper = shallow(
    <Movies 
    movies={[]}
    getCurrentMovieChars={getCurrentMovieChars}
    />)
    const mockEvent = { preventDefault() { }, target: {name: 'currentCharacters', value: ['char1', 'char2', 'char3'] } };
    const expected = ['char1', 'char2', 'char3']

    wrapper.instance().getCurrentMovieChars(mockEvent, mockEvent)

    expect(wrapper.state('currentCharacters')).toEqual(expected)
  })
});