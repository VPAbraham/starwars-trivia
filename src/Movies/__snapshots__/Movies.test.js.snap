// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should match the snapshot 1`] = `
<section
  className="section section--movie-list"
>
  <ul
    className="ul ul--movie-list"
  >
    <li
      className="li li--movie"
    >
      <MovieCard
        title="StarWars"
      />
    </li>
  </ul>
</section>
`;
